File: 2022/21/javascript/main.mjs
import fs from "node:fs";
import Resolver from "./resolver.mjs";

// --- Constants ---
const INPUT_FILE = "../input/input.txt";
const ROOT = "root";
const ME = "humn";

// --- Read and parse the input file ---
const text = fs.readFileSync(INPUT_FILE, "utf8");

// --- Puzzle 1 ---
let resolver = new Resolver(text);
let result = resolver.solveFor(ROOT);
console.log("1. Number that will the monkey root yell:", result.toLocaleString());

// --- Puzzle 2 ---
resolver = new Resolver(text);
resolver.updateForPuzzle2(ROOT, ME);
result = resolver.solveFor(ME);
console.log("2. Number that I yell to pass root's equality test:", result.toLocaleString());

File: 2022/21/python/main.py
from lib import Resolver

# --- Constants ---
INPUT_FILE = "../input/input.txt"
ROOT = "root"
ME = "humn"

# --- Puzzle 1 ---
resolver = Resolver(INPUT_FILE)
result = resolver.solve_for(ROOT)
print(f"1. Number that will the monkey root yell: {result:,d}")

# --- Puzzle 2 ---
resolver = Resolver(INPUT_FILE)
resolver.update_for_puzzle2(ROOT, ME)
result = resolver.solve_for(ME)
print(f"2. Number that I yell to pass root's equality test: {result:,d}")
